src/posix/platform/netif.cpp

```diff
@@ -1457,6 +1457,7 @@ void platformNetifInit(otInstance *aInstance, const char *aInterfaceName)
     mldListenerInit();
 #endif
 
+    otIp6SetReceiveFilterEnabled(aInstance, true);
```

Comment:
How about enhancing the otIp6SetReceiveFilterEnabled() API to a API which configures the detailed filter rules instead of just enable/disable?

Reply:
Currently, the changes introduced by this PR is just what we need, which fixes https://github.com/openthread/ot-br-posix/issues/491 . 
I think it's unclear whether or not an enhanced implementation like `otIp6SetReceiveFilterEnabled` is necessary for now, which also could unnecessarily increase code size. 

I prefer to enhance it when we really have the requirement in the future. 

Thoughts?

Reply:
wpantund has a option to enable/disable mapping RLOC/ALOC addresses, which means even when platform netif is not enabled, the messages may still need to be delivered to host IP stack. I'm fine with enhancing this later.